api/base/index.ts

```diff
@@ -89,7 +89,7 @@ export class SmartlingBaseApi {
         verb: string,
         uri: string,
         payload: Record<string, unknown> | string | FormData = null,
-        returnRawResponseBody = false,
+        returnRawResponseBody: boolean | 'buffer' = false,
```

Comment:
Maybe let's introduce enum and use it instead of having mixed type hint? Like:
```
enum ResponseBodyType {
    RAW_STRING,
    ARRAY_BUFFER,
    JSON,
}
```
and then use it as typehint like:
```
returnResponseBodyAs: ResponseBodyType = ResponseBodyType.JSON,
```
and make request code will look like:
```
...
switch (returnResponseBodyAs) {
    case ResponseBodyType.RAW_STRING:
        return response.text();

    case ResponseBodyType.ARRAY_BUFFER:
        return response.arrayBuffer();
}
...
```

Reply:
I will work on this